Mathlib/Algebra/Order/GroupWithZero/Range.lean

```diff
@@ -0,0 +1,71 @@
+/-
+Copyright (c) 2025 Filippo A. E. Nuccio. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Filippo A. E. Nuccio
+-/
+import Mathlib.Algebra.GroupWithZero.Range
+import Mathlib.Algebra.Order.GroupWithZero.WithZero
+
+/-! # The range of a MonoidWithZeroHom
+
+Given a `MonoidWithZeroHom` `f : A → B` whose codomain `B` is a `LinearOrderedCommGroupWithZero`,
+we provide some order properties of the `MonoidWithZeroHom.valueGroup₀` as defined in
+`Mathlib.Algebra.GroupWithZero.Range`.
+
+-/
+
+namespace MonoidWithZeroHom
+
+variable {A B F : Type*} [FunLike F A B]
+variable [MonoidWithZero A] [LinearOrderedCommGroupWithZero B]
+variable [MonoidWithZeroHomClass F A B] {f : F}
+
+open WithZero
+
+/-- The inclusion of `valueGroup₀ f` into `WithZero Bˣ` as an homomorphism of monoids with zero. -/
+@[simps!]
+def valueGroup₀_MonoidWithZeroHom : valueGroup₀ f →*₀ WithZero Bˣ :=
+  WithZero.map' (valueGroup f).subtype
+
+lemma valueGroup₀_MonoidWithZeroHom_strictMono :
+    StrictMono (valueGroup₀_MonoidWithZeroHom (f := f)) :=
+  map'_strictMono (Subtype.strictMono_coe _)
+
+/-- The inclusion of `valueGroup₀ f` into `WithZero Bˣ` as an order embedding. -/
+def valueGroup₀_OrderEmbedding : valueGroup₀ f ↪o WithZero Bˣ where
```

Comment:
You could combine the two into a `[OrderMonoidWithZeroHom](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Order/Hom/MonoidWithZero.html#OrderMonoidWithZeroHom)`

which I guess doesn't directly encode the StrictMono aspect

If you leave it as is, can you add information in the docstring please about which hom you prefer people use? The group hom, and rely on the strictMono aspect, or the order hom, and rely on the `mul` lemma?